js/src/utils.ts

```diff
@@ -43,15 +43,33 @@ export const divideBnToNumber = (numerator: BN, denominator: BN): number => {
   return quotient + rem.div(gcd).toNumber() / denominator.div(gcd).toNumber();
 };
 
+export const computeUiPrice = (market: Market, fp32Price: BN) => {
+  return roundUiAmount(
+    2 ** -32 *
+      10 ** (market.baseDecimals - market.quoteDecimals) *
+      fp32Price.toNumber()
+  );
+};
+
+export const computeUiSize = (market: Market, bnSize: BN) => {
+  return bnSize.toNumber() * 10 ** -market.baseDecimals;
+};
+
+export const roundUiAmount = (uiAmount: number) =>
+  Number(uiAmount.toPrecision(5));
```

Comment:
This is needed as the 2 ^ 32 division leaves weird rounding artifacts 5+ decimals out.